### PR TITLE
Adding support to consume latest stable images of ocs operator

### PR DESCRIPTION
**Pull request by afreen23**
- Plugin tests can consume the latest stable image
- We adopted the methodology of linking the private repo secret to all service accounts in order to be consumed by pods.
- The secret is created in `openshift-storage namespace` to be consumed by all pods for pulling their bundles
- The secret is created in `openshift-marketplace` in order to create a custom catalog source

Fixes https://issues.redhat.com/browse/RHSTOR-1835

Update:

Some issues that kept us blocking now fixed.
- In 4.8 a change in  obc spec  to use `spec.objectBucketName` instead of `spec.ObjectBucketName`
- backing store deletion gets flaky at times (seen generally as well irrespective of ocs version/build). There are a lot of resources deleted at once in a single exec command. It takes sometime hence to delete the resources. The default 30s timeout for which does not seems enough sometimes.
- cluster readiness checks for noobaa, ceph cluster readiness too before proceeding ahead to ensure the following tests won't fail which are dependent on ceph cluster and noobaa system availability.

Signed-off-by: Afreen Rahman <afrahman@redhat.com>

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 1]
```diff
@@ -1,106 +1,142 @@
-import * as _ from 'lodash';
+import { NS } from '../utils/consts';
+import { ServiceAccountKind } from 'public/module/k8s';
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:2]:
```suggestion
import { ServiceAccountKind } from '@console/public/module/k8s';
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 2]
```diff
@@ -1,106 +1,142 @@
...
+      subscribeToOperator();
+      cy.byTestID('view-installed-operators-btn').click();
+      cy.log('Wait for operator phase to be `Installing`');
+      cy.byLegacyTestID('item-filter').type(`${OCS_OP}`);
+      cy.byTestID('status-text', { timeout: 60000 }).should('have.text', 'Installing');
+      // Waiting for 30 seconds to make sure service accounts are loaded.
       // eslint-disable-next-line cypress/no-unnecessary-waiting
-      cy.wait(10000);
-      cy.byTestID('resource-status').contains('Ready', { timeout: 900000 });
+      cy.wait(30000);
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:132]:
```suggestion
      cy.wait(30000);
```
Poll for it.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/consts.ts [hunk 3]
```diff
@@ -108,3 +116,22 @@ export const OSD_SIZES_MAP = {
...
+    name: 'ocs-catalogsource',
+  },
+  spec: {
+    sourceType: 'grpc',
+    image: CATALOG.IMAGE,
+    secrets: [CATALOG.SECRET],
+    displayName: 'OpenShift Container Storage',
+    publisher: 'Red Hat',
+  },
+};
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/consts.ts:137]: Move this to mocks.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/consts.ts [hunk 4]
```diff
@@ -1,6 +1,14 @@
...
 export const NS = 'openshift-storage';
 
+export const PULL_SECRET_PATH = '/var/run/operator-secret/dockerconfig';
+
+export const CATALOG = {
+  NAMESPACE: 'openshift-marketplace',
+  SECRET: 'ocs-secret',
+  IMAGE: 'quay.io/rhceph-dev/ocs-registry:latest-stable-4.8',
+};
+
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/consts.ts:11]: Move this to mocks

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 5]
```diff
@@ -1,106 +1,142 @@
...
+const subscribeToOperator = () => {
+  cy.log('Search in Operator Hub');
+  cy.clickNavLink(['Operators', 'OperatorHub']);
+  cy.byTestID('search-operatorhub').type('Openshift Container Storage');
+  cy.byTestID('ocs-operator-ocs-catalogsource-openshift-marketplace', { timeout: 60000 }).click();
+  cy.log('Subscribe to OCS Operator');
+  cy.byLegacyTestID('operator-install-btn').click({ force: true });
+  cy.byTestID('Operator recommended Namespace:-radio-input').should('be.checked');
+  cy.byTestID('install-operator').click();
+};
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:37]: Move this to views.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 6]
```diff
@@ -1,106 +1,142 @@
...
+  cy.exec(`oc delete pods --all -n ${NS}`);
+};
+
+const checkMonitoring = () => {
+  cy.log(`Check monitoring enablement in ${NS} namespace`);
+  cy.exec(`oc get project ${NS} -o json`).then((res) => {
+    const obj = JSON.parse(res.stdout);
+    expect(obj.metadata.labels?.['openshift.io/cluster-monitoring']).toEqual('true');
+  });
+};
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:59]: Move this to views.

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 7]
```diff
@@ -1,106 +1,142 @@
...
 
-      cy.log(`Select ${mode}`);
-      cy.byTestID('Internal-radio-input').should('be.checked');
+const createInternalStorageCluster = (encrypted: boolean) => {
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:61]:
```suggestion
const createInternalStorageCluster = (encrypted: boolean, mode = 'Internal') => {
```

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts [hunk 8]
```diff
@@ -1,106 +1,142 @@
...
+      );
+  });
+
+  // Wait for the storage cluster to reach Ready
+  // Storage Cluster CR flickers so wait for 10 seconds
+  // Disablng until ocs-operator fixes above issue
+  // eslint-disable-next-line cypress/no-unnecessary-waiting
+  cy.wait(10000);
+  cy.byTestID('resource-status').contains('Ready', { timeout: 900000 });
+};
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:114]: Move to views.

**Review by bipuladh — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts (hunk 2 above)
Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/integration-tests-cypress/support/index.ts:132]: Actually. these service accounts are created by the operator.
if we poll it, then we need to hardcode their names and ensure all of them available.
In practice, we will get them instantly in few seconds after operator installation.
The 30 secs is for being on the safer side.

**Comment by afreen23** [comment]
Failing ceph tests are fixed here #8906

**Comment by afreen23** [comment]
In noobaa object bucket claims -> the object bucket name comes as `spec.objectBucketName` and not `spec.ObjectBucketName`.

**Comment by bipuladh** [comment]
/test ceph-storage-plugin
Checking for flakes

**Pull request merged**